### ISSUE TITLE
setIn doesn't always create arrays for numeric keys

### ISSUE THREAD

**Issue by adierkens** [guigrpa/timm#7]
I've noticed some discrepancies when using numeric keys w/ `setIn`

Consider the following:
```javascript
setIn(undefined, ['x', 0, 'y'], true)
// { x: { '0': { y: true } } }

setIn({}, ['x', 0, 'y'], true)
// { x: [ { y: true } ] }
```

I would expect both examples to result in the same outcome, but for some reason it decided to use `0` as a key for a new object, rather than an index in an array.

There are other use cases with similar outcomes:
```javascript
setIn(undefined, [0, 'y', 'z'], true)
// [ { y: { z: true } } ]

setIn(undefined, ['x', 0, 'y', 'z'], true)
// { x: { '0': { y: { z: true} } } }
```

Again, I would expect the second to produce an array, rather than an object.

**Comment by guigrpa** [comment]
Thanks for the report and PR! Released in v1.2.4

**Comment by adierkens** [comment]
Thanks for the quick release!
